pkg/reconcilermanager/controllers/rootsync_controller.go

```diff
@@ -779,14 +791,14 @@ func (r *RootSyncReconciler) validateRootSecret(ctx context.Context, rootSync *v
 	return validateSecretData(rootSync.Spec.Auth, secret)
 }
 
-func (r *RootSyncReconciler) upsertClusterRoleBinding(ctx context.Context, reconcilerRef types.NamespacedName) (client.ObjectKey, error) {
+func (r *RootSyncReconciler) upsertClusterRoleBinding(ctx context.Context, reconcilerRef types.NamespacedName, clusterRole string) (client.ObjectKey, error) {
 	crbRef := client.ObjectKey{Name: RootSyncPermissionsName(reconcilerRef.Name)}
 	childCRB := &rbacv1.ClusterRoleBinding{}
 	childCRB.Name = crbRef.Name
 
 	op, err := CreateOrUpdate(ctx, r.client, childCRB, func() error {
 		childCRB.OwnerReferences = nil
-		childCRB.RoleRef = rolereference("cluster-admin", "ClusterRole")
+		childCRB.RoleRef = rolereference(clusterRole, "ClusterRole")
 		childCRB.Subjects = addSubject(childCRB.Subjects, r.serviceAccountSubject(reconcilerRef))
```

Comment:
Or wait, are you saying we need to ignore existing bindings here, and set it to `addSubject(nil, ...)`? That's probably correct. That case should also have a test...

Reply:
Yeah since there is a 1:1 relationship between `RootSync` and `ClusterRoleBinding`, I agree that it seems correct to set this to a single subject.

This would handle the upgrade path in the case that a `RootSync` named `root-sync` exists, but requires user intervention otherwise. See my other comment for a proposal on how to handle the upgrade in the general case.